aiida/engine/processes/exit_code.py

```diff
@@ -8,49 +8,66 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """A namedtuple and namespace for ExitCodes that can be used to exit from Processes."""
-
-from collections import namedtuple
-
 from aiida.common.extendeddicts import AttributeDict
 
 __all__ = ('ExitCode', 'ExitCodesNamespace')
 
-ExitCode = namedtuple('ExitCode', ['status', 'message', 'invalidates_cache'])
```

Comment:
This is technically a change in interface, because `namedtuple` exposes all the methods of the underlying `tuple`. For example, one could do 

```
status, message, invalidates_cache = *ExitCode()
```

I'm not sure if any of these "accidental features" is used anywhere, though.

Reply:
Good point. Do you know of a way to make a simple class replicate the exact interface/behavior of a named tuple? If not, we might have to put this in `2.0`

Reply:
Indeed. Also this functionality is probably not super critical, depending on when a `2.0` release is scheduled.

Reply:
Behold, a thing of beauty:
```python
from collections import namedtuple

class ExitCode(namedtuple('_ExitCode', ['status', 'message', 'invalidates_cache'])):

    def __call__(self, **kwargs):
        pass

ExitCode.__new__.__defaults__ = (0, None, False)

ExitCode(0, 'bla')()
```

Reply:
![Yer-a-wizard](https://media.giphy.com/media/eax0rh3OERAYg/giphy.gif)